modules/log4j2/src/test/java/org/apache/ignite/logger/log4j2/Log4j2ConfigUpdateTest.java

```diff
@@ -147,6 +153,45 @@ private void checkConfigUpdate(Log4J2LoggerSupplier logSupplier) throws Exceptio
         Log4J2Logger.cleanup();
     }
 
+    /**
+     * Checks that Log4JLogger is updated after configuration file is changed.
+     *  @param logSupplier Function returning a logger instance to be tested.
+     */
+    @SuppressWarnings("ResultOfMethodCallIgnored")
+    private void checkDifferentInstances(Log4J2LoggerSupplier logSupplier) throws Exception {
+        Log4J2Logger.cleanup();
+
+        File infoCfgFile = new File(U.getIgniteHome(), LOG_CONFIG_INFO);
+        File debugCfgFile = new File(U.getIgniteHome(), LOG_CONFIG_DEBUG);
+        File logFile = new File(U.getIgniteHome(), LOG_DEST);
+
+        System.out.println("INFO config: " + infoCfgFile);
+        System.out.println("DEBUG config: " + debugCfgFile);
+        System.out.println("Log file: " + logFile);
+
+        if (logFile.delete())
+            System.out.println("Old log file was deleted.");
+
+        Log4J2Logger log = logSupplier.get(infoCfgFile);
+        assertTrue(log.toString().contains(infoCfgFile.getPath()));
+
+        log.info("Accepted info");
+        log.debug("Ignored debug");
+
+        log = logSupplier.get(debugCfgFile);
```

Comment:
Configuration of log is still infoCfgFile because cfg field is inited only once. That's why "Accepted debug" won't be logged